src/driver/multistage.cpp

```diff
@@ -90,6 +90,10 @@ TaskListStatus MultiStageDriver::Step() {
   TaskListStatus status;
   integrator->dt = tm.dt;
   for (int stage = 1; stage <= integrator->nstages; stage++) {
+    // Clear any initialization info. We should be relying
+    // on only the immediately preceding stage to contain
+    // reasonable data
+    pmesh->SetAllVariablesToInitialized();
```

Comment:
All that `SetAllVariablesToInitialized` does is to set `VariableState::initialized = true` for all variables. It doesn't change anything about the values in the field. 

The idea is that newly allocated fields on a block are set to zero but in some cases, users may want to set a different default value on allocation. This could be just a number other than zero or something more complicated that depends on the local state. If they call the task `InitNewlyAllocatedVars`, the variables with `VariableState::Initialized = false` get set to `VariableState::sparse_default_val` initially, and then the user methods for initialization the user has registered in various packages can search for where the fields are set to `sparse_default_val` and make whatever changes. (One possibility is that `VariableState::sparse_default_val` would be set to a nan_tag.)

But if the user doesn't add the task `InitNewlyAllocatedVars` and just assumes fields are initialized to zero, the initialized flags still need to be set to `true` at some point (although they are probably never being used).

Reply:
I see. To be honest, I did not get this (initially) from reading the docs. If you feel like it may be worth adding some info around that I won't object.

Reply:
At some point I will go through and update the developer sparse documentation to more clearly describe this.